pkg/types/ccip/gaspriceestimator.go

```diff
@@ -0,0 +1,47 @@
+package ccip
+
+import (
+	"context"
+	"math/big"
+	"time"
+)
+
+type GasPriceEstimator interface {
+	GasPriceEstimatorCommit
+	GasPriceEstimatorExec
+}
+
+type GasPriceEstimatorCommit interface {
+	CommonGasPriceEstimator
+
+	// Deviates checks if p1 gas price diffs from p2 by deviation options. Input prices should not be nil.
+	Deviates(p1 *big.Int, p2 *big.Int) (bool, error)
```

Comment:
Does this need to be an interface method? Feels like an odd calculation to do over an RPC call when it can be done prob 1000x faster locally.

Reply:
i had similar thoughts.

i started looking at the gas price estimator code in ccip to remove String and found some opportunities to refactor and make it cleaner independent of these methods.

personally, i would be ok with a TODO here referencing a ticket to clean this up. it's something i can pickup as part of implementing the grpc layer

Reply:
Fair enough!